From: Alessandro Ghedini <al3xbio@gmail.com>
Date: Sun, 1 May 2011 21:00:26 +0200
Subject: fix build of example and test

This patches fixes the build of the example and the test, by linking them to
the dynamic library instead of the static one.
---
 Makefile |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index b6b8f4f..f77db02 100644
--- a/Makefile
+++ b/Makefile
@@ -35,7 +35,7 @@ ifeq ($(uname_S),Darwin)
 else
   CFLAGS?=$(OPTIMIZATION) -fPIC -Wall -W -Wstrict-prototypes -Wwrite-strings $(ARCH) $(PROF)
   CCLINK?=-lm -pthread
-  LDFLAGS?=-L.
+  LDFLAGS?=-L. -Wl,-rpath,.
   DYLIBSUFFIX=so
   STLIBSUFFIX=a
   DYLIBNAME?=$(LIBNAME).$(DYLIBSUFFIX).$(VERSION)
@@ -94,7 +94,7 @@ hiredis-example-ae: example-ae.c adapters/ae.h $(STLIBNAME)
 endif
 
 hiredis-%: %.o $(STLIBNAME)
-	$(CC) -o $@ $(CCOPT) $(DEBUG) $(LDFLAGS) $(STLIBNAME) $<
+	$(CC) -o $@ $(CCOPT) $(DEBUG) $(LDFLAGS) -lhiredis $<
 
 test: hiredis-test
 	./hiredis-test
-- 
